nemo/src/execution/planning/plan_util.rs

```diff
@@ -34,170 +35,54 @@ use crate::{
-                            filter_classes[li].insert(right_variable);
-                        }
-                    },
-                    None => match right_index {
-                        Some(ri) => {
-                            filter_classes[ri].insert(left_variable);
-                        }
-                        None => {
-                            let mut new_set = HashSet::new();
-                            new_set.insert(left_variable);
-                            new_set.insert(right_variable);
-
-                            filter_classes.push(new_set);
-                        }
-                    },
-                }
-
-                continue;
-            }
-        }
-
-        // Case 2: Variable is compared to another variable or with a constant
-        // In this case, we need to update the `filter_assignments`
-
-        match &filter.rhs {
-            Term::Variable(right_variable) => {
-                if !variable_order.contains(right_variable) {
-                    continue;
-                }
-
-                let column_idx_left = *variable_order
-                    .get(left_variable)
-                    .expect("Loop iteration is skipped for unknown variables.");
-                let column_idx_right = *variable_order
-                    .get(right_variable)
-                    .expect("Loop iteration is skipped for unknown variables.");
-
-                let (column_idx_value, column_idx_bound, operation) =
-                    if column_idx_left > column_idx_right {
-                        (column_idx_left, column_idx_right, filter.operation)
-                    } else {
-                        (column_idx_right, column_idx_left, filter.operation.flip())
-                    };
-
-                let current_assignment = filter_assignments.entry(column_idx_value).or_default();
-
-                add_restriction(
-                    &operation,
-                    FilterValue::Column(column_idx_bound),
-                    &mut current_assignment.lower_bounds,
-                    &mut current_assignment.upper_bounds,
-                    &mut current_assignment.avoid_values,
-                );
+) -> Vec<ConditionStatement<DataValueT>> {
+    let mut result = Vec::new();
+
+    for constraint in constraints {
+        let some_variable = constraint
+            .variables()
+            .next()
+            .expect("A constraint should contain at least one variable");
+        let variable_type = variable_types
+            .get(some_variable)
+            .expect("Every variable should have a type.");
+
+        let (left, right) = constraint.terms();
+        let left_tree = termtree_to_arithmetictree(left, variable_order, variable_type);
+        let right_tree = termtree_to_arithmetictree(right, variable_order, variable_type);
+
+        let new_statement = match constraint.condition() {
+            Condition::Assignment(_, _) => unreachable!("Not a valid constraint"),
```

Comment:
"Not a valid constraint" seems a bit broad. Something along the lines of "Equality constraints are normalized into joins." (if this is even correct)?

Reply:
Not quite. Basically, a "constraint" is the same as a "condition" except for "Assignment" (which works slightly different from equality). I just laziliy reused the same structure instead of creating another enum with 90% of the same values.

I guess a nicer way to do this would be to define Constraints and then have 
```rs
enum Condition {
   Constraint(Constraint),
   Assignment(var, term)
}
```